include/Piece.h

```diff
@@ -33,6 +33,10 @@ namespace Chess {
             Chess::utilities::PieceType get_type() const { return type;};
             char get_symbol() const {return symbol;};
             bool get_had_moved() const { return had_moved;};
+            void set_had_move() { had_moved = true; };
```

Comment:
Penso volessi scrivere così

```suggestion
            void set_had_moved() { had_moved = true; };
```